src/Features/CSharp/Portable/CodeRefactorings/InvertIf/InvertIfCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,218 @@
+
+                    var notToken = logicalNotExpression.OperatorToken;
+                    var nextToken = logicalNotExpression.Operand.GetFirstToken(
+                        includeZeroWidth: true, includeSkipped: true, includeDirectives: true, includeDocumentationComments: true);
+
+                    var existingTrivia = SyntaxFactory.TriviaList(
+                        notToken.LeadingTrivia.Concat(
+                            notToken.TrailingTrivia.Concat(
+                                nextToken.LeadingTrivia)));
+
+                    var updatedNextToken = nextToken.WithLeadingTrivia(existingTrivia);
+
+                    return logicalNotExpression.Operand.ReplaceToken(
+                        nextToken,
+                        updatedNextToken);
+                }
+
+                case SyntaxKind.LogicalOrExpression:
+                {
+                    var binaryExpression = (BinaryExpressionSyntax)expression;
+                    result = SyntaxFactory.BinaryExpression(
+                        SyntaxKind.LogicalAndExpression,
+                        Negate(binaryExpression.Left, semanticModel, cancellationToken),
+                        SyntaxFactory.Token(
+                            binaryExpression.OperatorToken.LeadingTrivia,
+                            SyntaxKind.AmpersandAmpersandToken,
+                            binaryExpression.OperatorToken.TrailingTrivia),
+                        Negate(binaryExpression.Right, semanticModel, cancellationToken));
+
+                    return result
+                        .Parenthesize()
+                        .WithLeadingTrivia(binaryExpression.GetLeadingTrivia())
+                        .WithTrailingTrivia(binaryExpression.GetTrailingTrivia());
+                }
+
+                case SyntaxKind.LogicalAndExpression:
+                {
+                    var binaryExpression = (BinaryExpressionSyntax)expression;
+                    result = SyntaxFactory.BinaryExpression(
+                        SyntaxKind.LogicalOrExpression,
+                        Negate(binaryExpression.Left, semanticModel, cancellationToken),
+                        SyntaxFactory.Token(
+                            binaryExpression.OperatorToken.LeadingTrivia,
+                            SyntaxKind.BarBarToken,
+                            binaryExpression.OperatorToken.TrailingTrivia),
+                        Negate(binaryExpression.Right, semanticModel, cancellationToken));
+
+                    return result
+                        .Parenthesize()
+                        .WithLeadingTrivia(binaryExpression.GetLeadingTrivia())
+                        .WithTrailingTrivia(binaryExpression.GetTrailingTrivia());
+                }
+
+                case SyntaxKind.TrueLiteralExpression:
+                {
+                    var literalExpression = (LiteralExpressionSyntax)expression;
+                    return SyntaxFactory.LiteralExpression(
```

Comment:
Why not use the SyntaxGenerator helpers, such as [SyntaxGenerator.FalseLiteralExpression](http://source.roslyn.io/#Microsoft.CodeAnalysis.Workspaces/Editing/SyntaxGenerator.cs,29950d27013e228e,references)?